### DISCUSSION TITLE
Configure prisma to retry connections

### DISCUSSION THREAD

**Discussion by paymog** [prisma/prisma#19978] [Q&A] [answered]
Is there a way to configure Prisma to retry db connections? I ran into an issue over the weekend where there was instability in my postgres proxy which caused my api layer to fail to serve queries. This instability was transient and ideally I'd like to configure Prisma to retry connecting. Is that possible?

I looked through the [env var reference](https://www.prisma.io/docs/reference/api-reference/environment-variables-reference#proxy-environment-variables), [client api reference](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference), [schema reference](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference) and the [database reference](https://www.prisma.io/docs/reference/database-reference) and didn't see anything which might indicate this feature.

Is it possible to configure prisma to retry connecting to a database when the connection fails?

**Comment by nurul3101** [comment, marked as answer]
Hello @paymog 👋 

I was able to find a related GitHub Issue:

- https://github.com/prisma/prisma/issues/18071

It looks like you are running into the same issue. Can you please enable [jsonProtocol](https://github.com/prisma/prisma/issues/18095) and check if that fixes the issue? ([reference](https://github.com/prisma/prisma/issues/18071#issuecomment-1562581260))

**Reply by paymog** [reply]
I can definitely try that! I'll have to upgrade our client first though (currently on 4.2.1) so it might take some time to give this a solid test.

**Reply by paymog** [reply]
just to clarify, will prisma automatically retry connections when it encounters issues like the following when using the `jsonProtocol`?

```
Invalid `prisma.project.findUnique()` invocation: Can't reach database server at `proxy-graph-node.proxy-cn9cukuadcb4.us-west-2.rds.amazonaws.com`:`5432` Please make sure your database server is running at `proxy-graph-node.proxy-cn9cukuadcb4.us-west-2.rds.amazonaws.com`:`5432`.
```

**Reply by nurul3101** [reply]
I would recommend upgrading to the latest version of Prisma i.e. `4.16.2`, and checking if the issue gets resolved.

> will prisma automatically retry connections when it encounters issues like the following when using the `jsonProtocol`?

Because the [comment](https://github.com/prisma/prisma/issues/18071#issuecomment-1562581260) mentions that in the latest version of Prisma, this issue should be resolved even if `jsonProtocol` is not enabled.

**Reply by nurul3101** [reply]
Did you get a chance to try out version `4.16.2`?

**Reply by paymog** [reply]
Yup, it seems to be working! Hopefully the issue we ran into before with AWS infra doesn't crop up again 🙂
